core/src/main/java/org/apache/accumulo/core/client/impl/TableOperationsImpl.java

```diff
@@ -1208,11 +1207,20 @@ public void online(String tableName) throws AccumuloSecurityException, AccumuloE
   @Override
   public void online(String tableName, boolean wait) throws AccumuloSecurityException, AccumuloException, TableNotFoundException {
     checkArgument(tableName != null, "tableName is null");
+
+    long cacheResetCount = Tables.getCacheResetCount();
+
     String tableId = Tables.getTableId(context.getInstance(), tableName);
 
     /**
      * ACCUMULO-4574 if table is already online return without executing fate operation.
      */
+
+    // getTableId may have side effect of clearing zookeeper cache - no need to clear again.
+    if (cacheResetCount == Tables.getCacheResetCount()) {
+      Tables.clearCacheByPath(context.getInstance(), Constants.ZTABLES + "/" + tableId + Constants.ZTABLE_STATE);
```

Comment:
I have no issue with incorporating this suggestion - leaning towards a getTableState state method with a boolean - however,  a question: Are there other places that would benefit from having the general method? I was assuming that with by passing in a generic path the method would be more general and could be used elsewhere.  Or, are you suggesting that we keep both? I have no strong opinion either way.